scenario_player/utils/token.py

```diff
@@ -15,456 +17,249 @@
+    """
+    given_token_address = token_proxy.address
+    user_deposit_token_address = userdeposit_proxy.token_address()
+
+    if user_deposit_token_address != given_token_address:
+        raise ValueError(
+            f"The allowance for the user deposit contract must be increase on "
+            f"the corresponding token. Given token: {given_token_address} user "
+            f"deposit token: {user_deposit_token_address}."
         )
-        return self.checksum_address, self.deployment_block
 
+    current_allowance = token_proxy.allowance()
 
-class UserDepositContract(Contract):
-    """User Deposit Contract wrapper for scenario runs.
+    if minimum_allowance > current_allowance:
+        return token_proxy.allowance(maximum_allowance, userdeposit_proxy.address)
 
-    Takes care of:
+    return None
 
-        - Minting tokens for nodes on the UDC
-        - Updating the allowance of nodes
-    """
 
-    def __init__(self, scenario_runner, contract_proxy, token_proxy):
-        super().__init__(scenario_runner, address=contract_proxy.address)
-        self.contract_proxy = contract_proxy
-        self.token_proxy = token_proxy
-        self.tx_hashes = set()
-
-    @property
-    def ud_token_address(self) -> ChecksumAddress:
-        return to_checksum_address(self.token_proxy.address)
-
-    @property
-    def allowance(self):
-        """Return the currently configured allowance of the UDToken Contract."""
-        return self.token_proxy.functions.allowance(
-            self._local_rpc_client.address, self.address
-        ).call()
-
-    @property
-    def balance(self):
-        """Proxy the balance call to the UDTC."""
-        return self.token_proxy.functions.balanceOf(self.ud_token_address).call()
-
-    def effective_balance(self, at_target):
-        """Get the effective balance of the target address."""
-        return self.contract_proxy.functions.effectiveBalance(at_target).call()
-
-    def total_deposit(self, at_target):
-        """"Get the so far deposted amount"""
-        return self.contract_proxy.functions.total_deposit(at_target).call()
-
-    def mint(
-        self, target_address, required_balance=None, max_fund_amount=None, **kwargs
-    ) -> Union[TransactionHash, None]:
-        """The mint function isn't present on the UDC, pass the UDTC address instead."""
-        return super().mint(
-            target_address,
-            required_balance=required_balance,
-            max_fund_amount=max_fund_amount,
-            contract_address=self.ud_token_address,
-            **kwargs,
-        )
+def userdeposit_maybe_deposit(
+    user_deposit_proxy: UserDeposit,
+    target_address: ChecksumAddress,
+    minimum_effective_deposit: int,
+    maximum_funding: int,
+) -> Optional[TransactionHash]:
+    """Make a deposit at the given `target_address`.
+
+    The amount of tokens depends on the scenario definition's settings.
 
-    def update_allowance(self) -> Tuple[Optional[TransactionHash], int]:
```

Comment:
this code was condition, moved out of the method into a plain function named `userdeposit_maybe_increase_allowance`